documentation/source/PhaseCallbacks.md

```diff
@@ -0,0 +1,280 @@
+# Phase Callbacks in SG
+
+Integrating your own code into an already existing training pipeline can draw much effort on the user's end.
+To tackle this challenge, a list of callables triggered at specific points of the training code can be passed through `phase_calbacks_list` inside `training_params` when calling `Trainer.train(...)`.
+
+SG's `super_gradients.training.utils.callbacks` module implements some common use cases as callbacks:
+
+    ModelConversionCheckCallback
+    DeciLabUploadCallback
+    LRCallbackBase
+    EpochStepWarmupLRCallback
+    BatchStepLinearWarmupLRCallback
+    StepLRCallback
+    ExponentialLRCallback
+    PolyLRCallback
+    CosineLRCallback
+    FunctionLRCallback
+    LRSchedulerCallback
+    DetectionVisualizationCallback
+    BinarySegmentationVisualizationCallback
+    TrainingStageSwitchCallbackBase
+    YoloXTrainingStageSwitchCallback
+
+For example, the YoloX's COCO detection training recipe uses `YoloXTrainingStageSwitchCallback` to turn off augmentations and incorporate L1 loss starting from epoch 285:
+
+`super_gradients/recipes/training_hyperparams/coco2017_yolox_train_params.yaml`:
+
+```yaml
+
+max_epochs: 300
+...
+
+loss: yolox_loss
+
+...
+
+phase_callbacks:
+  - YoloXTrainingStageSwitchCallback:
+      next_stage_start_epoch: 285
+...
+```
+
+Another example is how we use `BinarySegmentationVisualizationCallback` to visualize predictions during training in our [Segmentation Transfer Learning Notebook](https://bit.ly/3qKwMbe):
+
+
+## Integrating Your Code with Callbacks
+
+Integrating your code requires implementing a callback that `Trainer` would trigger in the equivalent areas inside SG's training pipeline.
```

Comment:
instead of "equivalent areas", "proper phases"

Reply:
fixed.

Reply:
fixed in latest commit.